### ISSUE TITLE
Main site pricing page: Font size too small [a11y]

### ISSUE THREAD

**Issue by GrahamTheDevRel** [supabase/supabase#15356]
## Describe the bug

On the pricing page: (https://github.com/supabase/supabase/blob/master/apps/www/pages/pricing/index.tsx) the font size is too small on each of the feature lists and in various other places.

## Expected behaviour

The font size should be at least 16px equivalent (currently 12px equivalent). There is no WCAG guidance on font size, but this is the generally accepted guidance for minimum font size for accessibility.

## Additional context

The fix is to change the font size from `text-xs` to `text-s` and adjust any layout issues that may occur. 

* note - this is a site wide issue, but I was only reviewing the Pricing page where it seemed particularly prevalent.

**Comment by Arindam200** [comment]
Can I Work on this?

**Comment by Birdie13-stack** [comment]
I will like to work on this

**Comment by MildTomato** [comment]
> The font size should be at least 16px equivalent (currently 12px equivalent). There is no WCAG guidance on font size, but this is the generally accepted guidance for minimum font size for accessibility.

Hey @GrahamTheDevRel - firstly - since this is the first reply to your a11y issues, i love the enthusiasm. We have a big commitment ongoing to replace a lot of UI now with shadcn based components across all our website. 

I just spent the last few days setting up new UI components a lot of it was to try and really globally fix a lot of the a11y issues across all our apps. Generally I prefer the approach of replacing everything with a robust system than playing whack-a-mole on specific issues, mainly because there are **_a lot of them_**, and also because a lot of components need to be literally replaced (rather than an import swap).

anyway, about the font size.

we're generally in agreement internally that we 'hacked' our tailwind a bit by not setting a correct base font size. So then tailwind `text-xs` was used everywhere and ended up becoming a bit of an injoke.

The correct setup would be to setup the base font size correctly, and then use 'font-base' as, well, the base font size.

It might be slightly controversial but we'll probably settle on something like 13px or 14px as the base font size.
We're aware of font sizes in dev tools like vscode which are generally on the small side (14px), and we are perhaps a bit biased with recent trends towards small font sizes ([Linear](https://linear.app/) being an obvious one).

It would be interesting to see what others thought though, we can review the feedback in here.

**Comment by MildTomato** [comment]
I forgot to mention in the last comment.

It would make sense if we actually supported some a11y options like text sizes/spacing, especially on things like data tables. It's not really solvable overnight but it's something I would personally like to see supported in future.

**Comment by GrahamTheDevRel** [comment]
@MildTomato 14px, while small, isn't too far off as you are using REM units I suppose (and not setting the base font size in PX), so it should respect browser preferences and user style sheets.

As with anything, it is a compromise. 

If you are able to persuade everyone to 16px then that is a win, an accessibility audit would still flag 14px as will most automated testers, but if 14px is the best you can do without a massive redesign then I get it! 

I agree on not playing whack-a-mole so I will hang back on raising further issues, but am happy to help if you need me on the new components!

**Comment by MildTomato** [comment]
@GrahamTheDevRel we are going to really revamp our theming system, I think one of the things we should make sure there is a css variable for the base font size (for body), and then another one for data (for data tables, terminal outputs etc), and then we have something we can control easily - maybe allow developers to independently control.

IMO I really don't think having a minimum of 16px font size everywhere is particular great.
Even https://w3.org doesn't even use min of 16px everywhere, they have 15px on their side menus.

Maybe some examples of popular sites not using 16px:

- GitHub body font is 14px. (The text-size of this very comment in GitHub is 14px. oddly though, the css variables has a base - variable of 16px, but they don't seem to use it much)
- Stackoverflow base is 13px. (They do push to 15px for thread comments, nav/links are 13px)
- VSCode default is 14px
- Vercel is 14px _(to be fair, their actual base is 16px, but most of the text sizes are lowered for body)_.

I think we would be doing a disservice to developers to move away from what seems like a it's own standard. 

But, I still do agree it is currently too small in a lot of places. 

Our font is also not helping either as it doesn't read well. Even though most of our dashboard is 14px, it would look quite different with something like the font Inter. So some of the issue is just to do with using Circular font I think, maybe we are using the wrong weight for that font size.

**Comment by MildTomato** [comment]
The example you cited text-xs is certainly too small. 13px should probably be the bottom size.

<img width="690" alt="Screenshot 2023-06-27 at 12 28 28 PM" src="https://github.com/supabase/supabase/assets/8291514/131b1354-58b1-4381-b766-39b5f7456170">

@Arindam200 

maybe a good win here is to manually change text-xs to be 13px for now. I can have a look later as well, see what we can do to make sure we're not doing anything extreme.

**Comment by GrahamTheDevRel** [comment]
Agree with everything you are saying, as I said, compromises are to be expected!

Also bear in mind I am always going for "perfect" and I always design with 16px in mind...Because of that I have far fewer restrictions than you do so I can afford to be evangelical about things! 

I think your proposed solutions are certainly good enough and thanks for taking the time to explain your thought process in detail!
